pkg/handlers/health.go

```diff
@@ -0,0 +1,46 @@
+package handlers
+
+import (
+	"net/http"
+
+	"github.com/Ubivius/microservice-friendslist/pkg/data"
+)
+
+// LivenessCheck determine when the application needs to be restarted
+func (relationshipHandler *RelationshipsHandler) LivenessCheck(responseWriter http.ResponseWriter, request *http.Request) {
+	log.Info("LivenessCheck")
+	responseWriter.WriteHeader(http.StatusOK)
+}
+
+//ReadinessCheck verifies that the application is ready to accept requests
+func (relationshipHandler *RelationshipsHandler) ReadinessCheck(responseWriter http.ResponseWriter, request *http.Request) {
+	log.Info("ReadinessCheck")
+
+	readinessProbeMicroserviceUser := data.MicroserviceUserPath + "/health/ready"
+	readinessProbeMicroserviceTextChat := data.MicroserviceTextChatPath + "/health/ready"
+
+	_, errMicroserviceUser := http.Get(readinessProbeMicroserviceUser)
```

Comment:
http.Get is blocking in go. We could do one, then check the error then check the other. We could use err in all three cases if we do it this way and it will be more efficient if the first service we do the request on is down.

Reply:
True